src/autocomplete/TopicAutocomplete.js

```diff
@@ -17,23 +17,19 @@ const styles = createStyleSheet({
   },
 });
 
-type SelectorProps = $ReadOnly<{|
-  topics: string[],
-|}>;
-
 type Props = $ReadOnly<{|
   narrow: Narrow,
   isFocused: boolean,
   text: string,
   onAutocomplete: (name: string) => void,
-
-  dispatch: Dispatch,
-  ...SelectorProps,
 |}>;
 
-class TopicAutocomplete extends PureComponent<Props> {
-  componentDidMount() {
```

Comment:
I think this conversion is not actually NFC -- in fact I think it fixes a bug!

Namely, in the existing code we do a `fetchTopicsForStream(narrow)` here when the component first mounts. If `narrow` changes, we won't fetch topics again for the new stream.

Now, when `TopicAutocomplete` appears in `ComposeBox`, this is only a latent bug: `narrow` is the narrow we're viewing, and that only changes by navigation, making new components.

But it also appears in `ShareToStream`. I believe this means we had a visible bug there, where if you
* focused the topic input,
* then edited the stream input,
* and focused the topic input again,

we'd never go and fetch topics for the new stream, and so we'd populate the autocomplete only with whatever topics we happened to already know about for that stream.

With this change, we make a new fetch when the narrow changes. Because the fetch goes to Redux, that doesn't introduce any issue of being out of sync by the time the fetch returns. And the magic of `useEffect` and its explicit dependencies (and ESLint's ability to check the dependencies for completeness) means that it'd actually be rather harder now to write this code *with* this sort of bug than without it.

This commit is otherwise simple enough that I don't think there's a need for a separate commit for this fix -- just the commit message should reflect that it's this bugfix.

Reply:
Great! 😄